FNF_MissionTemplate.VR/server/damage/fn_handleDamage.sqf

```diff
@@ -0,0 +1,46 @@
+/*
+ * Author: tanaKa
+ * Function used to handle random damage on vehicle destruction
+ * for FNF
+ *
+ * Params: 
+ * 0: vehicle <OBJECT>
+ *
+ * Return value:
+ * N/A
+ *
+ * Usage: 
+ * [vehicle] call PHX_fnc_handleDamage
+ *
+ * Public: no
+ */
+params ["_vehicle"];
+
+{
+	_kill = random [1,50,100];
+
+	if (_kill > 69) then {
+		_bodyPart = selectRandom ["Head", "Body", "LeftArm", "RightArm", "LeftLeg", "RightLeg"];
+		_damage = 1; // full damage
+		_type = selectRandom ["bullet", "vehiclecrash", "collision", "grenade", "explosive", "shell", "backblast", "stab", "unknown"];
+		[_x, _damage, _bodyPart, _type] call ace_medical_fnc_addDamageToUnit;
+		_x setDamage 1; // kill player
+		systemChat format ["DEBUG: %1 damage added to %2 with a %3 value with _kill set to %4",_type,_bodypart,_damage,_kill];
+	} else {
+		_damage = selectRandom [0,0,0,0,0,0.2,0.3,0.4]; //small chance to be slightly injured in a limb
+		if (_damage > 0) then {
+			_bodyPart = selectRandom ["LeftArm", "RightArm", "LeftLeg", "RightLeg"];
+			_type = selectRandom ["bullet", "vehiclecrash", "collision", "grenade", "explosive", "shell", "backblast", "stab", "unknown"];
+			[_x, _damage, _bodyPart, _type] call ace_medical_fnc_addDamageToUnit;
+			systemChat format ["DEBUG: %1 damage added to %2 with a %3 value with _kill set to %4",_type,_bodypart,_damage,_kill];
+			uiSleep 2;
+			waitUntil {(getPos _vehicle) select 2 isEqualTo 0 && speed _vehicle isEqualTo 0};
```

Comment:
Z pos will never be exactly 0, so this will never return true. Likely good enough to just use the speed of the vic and eschew the z pos check. Even with the speed check, there can be situations that prevent the speed from ever reaching exactly 0 (Arma physics, boats, probably other situations), so a less-than check of the abs value of the speed is more ideal than a flat isEqualTo 0 -- so if some physics interaction is occurring that's keeping the speed at 0.001, etc., the function will still be able to progress.

Reply:
Also timeouts are nice for things like this, so if something does occur that's not accounted for in the waitUntil, this thread won't remain active forever.

Reply:
Resolved with https://github.com/FridayNightFight/FNF/commit/dd8f3ca0d2588cafad474784d2b85beb3188c584